drivers/metal/metal_objects.mm

```diff
@@ -597,37 +597,40 @@
 	MDRenderPass const &pass = *render.pass;
 	MDSubpass const &subpass = render.get_subpass();
 
-	// First determine attachments that should be cleared.
-	LocalVector<RDD::AttachmentClear> clears;
-	clears.reserve(subpass.color_references.size() + /* possible depth stencil clear */ 1);
+	uint32_t ds_index = subpass.depth_stencil_reference.attachment;
+	bool clear_depth = (ds_index != RDD::AttachmentReference::UNUSED && pass.attachments[ds_index].shouldClear(subpass, false));
+	bool clear_stencil = (ds_index != RDD::AttachmentReference::UNUSED && pass.attachments[ds_index].shouldClear(subpass, true));
 
-	for (uint32_t i = 0; i < subpass.color_references.size(); i++) {
+	uint32_t color_count = subpass.color_references.size();
+	uint32_t clear_count = color_count + (clear_depth || clear_stencil ? 1 : 0);
+	if (clear_count == 0) {
+		return;
+	}
+
+	RDD::AttachmentClear *clears = ALLOCA_ARRAY(RDD::AttachmentClear, clear_count);
+	uint32_t clears_idx = 0;
```

Comment:
This change didn't fix the bug, but is an optimisation to avoid allocating a `LocalVector` on the heap, as the number of clears will be 10 or less (max 8 for colour buffers, 1 depth and 1 stencil)